thingsboard_gateway/storage/sqlite/sqlite_event_storage.py

```
@@ -1,113 +1,352 @@
+        except Exception as e:
+            self.__log.debug("Close_db error: %s", e)
+        self.__write_database.db.close()
+
+    def __rotate_read_database(self):
+        self.__log.debug("Rotating read database")
+        all_files = self.__sort_db_files_list()
+        if len(all_files) > 1:
+            self.__start_read_database(all_files[0])
+        else:
+            self.__log.trace("Only one DB left, reusing write DB")
+            self.__read_database = self.__write_database
+            self.__read_database.should_read = True
+
+    def __check_and_handle_max_db_count(self) -> bool:
+        count = len(self.__sort_db_files_list())
+        if count >= self.__settings.max_db_amount:
+            self.__is_max_db_amount_reached = True
+            self.__log.warning("Max DB count (%d) reached—dropping writes", count)
+            return True
+        if self.__is_max_db_amount_reached:
+            self.__log.debug("DB count cleared, creating new write DB")
+            self.__is_max_db_amount_reached = False
+            self.__prepare_new_db_configuration()
+            self.__start_write_database()
+        return False
+
+    def __rotate_after_read_completion(self):
+        self.__log.debug("Rotate after read completion")
+        self.__delete_db_files(self.__read_database.settings.data_folder_path)
+        self.delete_time_point = 0
+        self.__rotate_read_database()
+        self.__check_and_handle_max_db_count()
 
     def event_pack_processing_done(self):
-        self.__log.trace("Batch processing done, processing time: %i milliseconds",
-                        int((monotonic() - self.__event_pack_processing_start) * 1000))
+        self.__log.trace(
+            "Batch done in %d ms",
+            int((monotonic() - self.__event_pack_processing_start) * 1000),
+        )
         if not self.stopped.is_set():
-            self.delete_data(self.delete_time_point)
+            try:
```

It should raise the error here, to handle it in the upper level.

---

Fixed